File: Python/5kyu/FirstNonRepeatingCharacter.py
def first_non_repeating_letter(string):
    result = [s for s in string if string.lower().count(s.lower()) == 1]

    if result:
        return result[0]

    return ""


def first_non_repeating_letter_kata_1(string):
    string_lower = string.lower()

    for i, letter in enumerate(string_lower):
        if string_lower.count(letter) == 1:
            return string[i]

    return ""


print(first_non_repeating_letter('moonmen'))

File: JavaScript/5kyu/FirstNonRepeatingCharacter.js
function firstNonRepeatingLetter(s) {
  for (var i in s) {
    if (s.match(new RegExp(s[i], "gi")).length == 1) {
      return s[i];
    }
    s.split("").map((c) => c);
  }

  return "";
}

function firstNonRepeatingLetter_kata_1(s) {
  return (
    s.split("").find((c) => s.match(new RegExp(c, "gi")).length == 1) || ""
  );
}

var result = firstNonRepeatingLetter_kata_1("stress");
console.log(result);
